src/requirements/specialize.ts

```diff
@@ -0,0 +1,40 @@
+import { includesWithSubRequirements } from './data/checkers-common';
+import { CollegeOrMajorRequirement, Major } from './types';
+
+/**
+ * Many majors often "specialize" the requirements of their respective college(s)
+ * `specialized(collegeReqs, majors)` is an array of `collegeReqs` having been
+ * specialized according to `majors`
+ *
+ * @param collegeReqs the base set of requirements for the college
+ * @param majors the set of majors to specialize `collegeReqs` for
+ * @returns An array of `collegeReqs` having been specialized for `majors`
+ */
+export function specialized<R extends RequirementCommon>(
+  collegeReqs: readonly R[],
+  majors: readonly Major<R>[]
+) {
+  const allSpecializations = majors.flatMap(({ specializations }) => specializations ?? []);
+  const excluded = new Set(allSpecializations.map(({ name }) => name));
+  const filteredReqs = collegeReqs.filter(({ name }) => !excluded.has(name));
+  return [...filteredReqs, ...allSpecializations];
+}
```

Comment:
Ideally, we would like to preserve the original order of requirements in the college, so maybe:

```typescript
export function specialized<R extends RequirementCommon>(
  collegeReqs: readonly R[],
  majors: readonly Major<R>[]
) {
  const allSpecializations = majors.flatMap(({ specializations }) => specializations ?? []);
  const specializationMap = new Map(allSpecializations.map((it) => [it.name, it]));
  return collegeReqs.map((it) => specializationMap.get(it.name) ?? it);
}
```